src/radix_connect/mobile/client.rs

```diff
@@ -1,65 +1,189 @@
-// use super::relay_service::Service as RelayService;
+use std::borrow::BorrowMut;
+
 use super::deep_link_parsing::*;
+use super::relay_service::Service as RelayService;
+use super::relay_service::WalletInteractionTransport;
 use crate::prelude::*;
+use hex::ToHex;
+use std::sync::RwLock;
+
+/// The Radix Connect Mobile client that handles the interaction with dApps on mobile through deepLinks.
+#[derive(uniffi::Object)]
+pub struct RadixConnectMobile {
+    /// The transport to be used to send back the response for the WalletInteraction
+    wallet_interactions_transport: Arc<dyn WalletInteractionTransport>,
+    /// The storage to be used to store the sessions established with dApps.
+    session_storage: Arc<dyn SessionStorage>,
```

Comment:
I think `SessionStorage` and `Session` might benefit from being renamed into `RCMSessionStorage` and `RCMSession`?

Reply:
`SessionStorage` could be, but `Session` is internal to the RadixConnectMobile module.